providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -1332,6 +1345,17 @@ static void *ec_gen(void *genctx, OSSL_CALLBACK *osslcb, void *cbarg)
 
     if (gctx->ecdh_mode != -1)
         ret = ret && ossl_ec_set_ecdh_cofactor_mode(ec, gctx->ecdh_mode);
+#ifdef FIPS_MODULE
+    /* Pairwise consistency test */
+    if ((gctx->selection & OSSL_KEYMGMT_SELECT_KEYPAIR) != 0
```

Comment:
Doesn't ec_generate_key() already do a PCT?

Reply:
> crypto/ec/ec_key.c:ecdsa_keygen_pairwise_test()
> As also required by CAVP FAQ GEN.2, this PCT shall perform hash+sign digital signature operations using an atomic function i.e., cannot rely on ECDSA_do_sign and ECDSA_do_verify which do not perform the hashing.

Reply:
So my patch is wrong, and instead it should be fixing ecdsa_keygen_pariwise_test to hash a message and sign it. Rather than signing a digest of zeros.

Reply:
The original patch was written the way it uses the combined callbacks (so that's why some functions stopped being static). I don't remember whether the pairwise test was called at that moment.

Reply:
Ack, but will make changes to ec_generate_key in a separate PR.